### PR TITLE
feat: delete user profile

### PR DESCRIPTION
**Pull request by olugbenga1**
# What does this PR do?
This PR implements an endpoint for deleting user profile:
* Added DELETE /api/v1/profile/:id Endpoint: Deletes a user's profile
* Authentication: Ensures only authenticated users can access these endpoints.
* Authorization: Ensures only users or admin can access this route.
* Error Handling: Returns specific HTTP status codes and messages based on various scenarios.

# Checklist:
- [x] Implemented DELETE /api/v1/profile/:id to delete the current user's profile settings.
- [x] Implemented an endpoint to delete user profile settings.
- [x] Return 200 OK on successful deletion.
- [x] Return 401 Unauthorized if the user is not authenticated.
- [x] Tested endpoint functionality to ensure it correctly handles various scenarios.
- [x] Added appropriate HTTP status codes and response messages for edge cases.
- [x] Added unit tests for the deleteProfile() method
- [x] Updated API documentation to reflect the new endpoint.

# Response Examples
**Success**

```json
{
    "status": "success",
    "message": "User profile deleted successfully"
}
```

# Link to Issue
Link to this issue: [FEAT Create API Endpoints to Delete User Profile](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/167)

### CODE REVIEW

File: src/modules/profile/profile.service.ts [hunk 1]
```diff
@@ -65,6 +65,36 @@ export class ProfileService {
         data: updatedProfile,
       };
 
+      return responseData;
+    } catch (error) {
```
Comment (Cobidex) [inline src/modules/profile/profile.service.ts:69]: we have an interceptor, no need for try catch

File: src/modules/profile/profile.service.ts [hunk 2]
```diff
@@ -65,6 +65,36 @@ export class ProfileService {
...
+      }
+
+      await this.profileRepository.softDelete(userProfile.id);
+
+      const responseData = {
+        message: 'Profile successfully deleted',
+      };
+
       return responseData;
     } catch (error) {
```
Comment (Cobidex) [inline src/modules/profile/profile.service.ts:99]: no need for try / catch

File: src/modules/profile/profile.service.ts [hunk 3]
```diff
@@ -65,6 +65,36 @@ export class ProfileService {
...
+      });
+
+      if (!userProfile) {
+        throw new NotFoundException('Profile not found');
+      }
+
+      await this.profileRepository.softDelete(userProfile.id);
+
+      const responseData = {
+        message: 'Profile successfully deleted',
```
Comment (Cobidex) [inline src/modules/profile/profile.service.ts:95]: Use constants for your error and response messages

**Review by Cobidex — APPROVED** [review]

**Review by TheCodeGhinux — APPROVED** [review]
